### ISSUE TITLE
Need a better char address for Unicode errors

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1524]
```
#!python

E:\pymove3d-en>make.bat html
Running Sphinx v1.3a0
loading translations [en]... done
loading pickled environment... failed: [Errno 2] No such file or directory: 'E:\\pymove3d-en\\_build\\doctrees\\environment.pickle'
loading intersphinx inventory from http://pymove3d.sudile.com/objects.inv...
building [html]: targets for 59 source files that are out of date
updating environment: 59 added, 0 changed, 0 removed
reading sources... [ 45&#37;] stations/blender-game-engine/c_mainloop              pinge
Encoding error:
'utf-8' codec can't decode byte 0xbb in position 760: invalid start byte
The full traceback has been saved in c:\temp\sphinx-err-dkpxr0.log, if you want to report the issue to the developers.
```

```
#!python
# Sphinx version: 1.3a0
# Python version: 3.3.4
# Docutils version: 0.10 release
# Jinja2 version: 2.6
# Loaded extensions:
#   sphinx.ext.todo from C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\ext\todo.py
#   sphinx.ext.ifconfig from C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\ext\ifconfig.py
#   sphinx.ext.extlinks from C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\ext\extlinks.py
#   sphinx.ext.intersphinx from C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\ext\intersphinx.py
#   sphinx.ext.mathjax from C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\ext\mathjax.py
#   sphinx.ext.pngmath from C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\ext\pngmath.py
Traceback (most recent call last):
  File "C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\cmdline.py", line 253, in main
    app.build(force_all, filenames)
  File "C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\application.py", line 223, in build
    self.builder.build_update()
  File "C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\builders\__init__.py", line 214, in build_update
    'out of date' &#37; len(to_build))
  File "C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\builders\__init__.py", line 234, in build
    purple, length):
  File "C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\builders\__init__.py", line 134, in status_iterator
    for item in iterable:
  File "C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\environment.py", line 474, in update_generator
    self.read_doc(docname, app=app)
  File "C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\environment.py", line 621, in read_doc
    pub.publish()
  File "C:\Python33\lib\site-packages\docutils-0.10-py3.3.egg\docutils\core.py", line 217, in publish
    self.settings)
  File "C:\Python33\lib\site-packages\docutils-0.10-py3.3.egg\docutils\readers\__init__.py", line 71, in read
    self.input = self.source.read()
  File "C:\Python33\lib\site-packages\sphinx-1.3a0dev_20140728-py3.3.egg\sphinx\environment.py", line 601, in read
    data = FileInput.read(self_)
  File "C:\Python33\lib\site-packages\docutils-0.10-py3.3.egg\docutils\io.py", line 265, in read
    data = self.source.read()
  File "C:\Python33\lib\codecs.py", line 301, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
  File "C:\Python33\lib\encodings\utf_8_sig.py", line 69, in _buffer_decode
    return codecs.utf_8_decode(input, errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbb in position 760: invalid start byte
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1524
- Originally reported by: anatoly techtonik
- Originally created at: 2014-07-28T11:13:55.524

**Comment by shimizukawa** [comment]
_From anatoly techtonik on 2014-07-28 09:20:54+00:00_

Note that Sphinx 1.2 works ok in this case.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-07-29 05:52:05+00:00_

Please attache a minimal rst source to reproduce it.

I can't reproduce the behavior with python3.3 and:
- utf-8 rst file that contain non-ascii strings
- utf-8-sig rst file (utf-8 + bom) that contain non-ascii string

**Comment by shimizukawa** [comment]
_From anatoly techtonik on 2014-07-29 10:37:27+00:00_

```
hg clone https://bitbucket.org/pkoppatz/pymove3d-en
cd pymove3d-en
hg up -r 115
make html
```

One example file is attached.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-07-29 17:47:57+00:00_
- Fix: Undecodable source reading cause UnicodeDecodeError instead of emitting warnings. Thanks to anatoly techtonik. Closes #1524

→ <<cset 6630255a109d1707868ee13f811f1561d1df3f2d>>
